src/redux/slices/user.ts

```diff
@@ -421,6 +425,66 @@ export const getTokenInfo = createAsyncThunk(
   }
 );
 
+export const getCollectionInfo = createAsyncThunk(
+  'user/getCollectionInfo',
+  async (
+    {
+      collection,
+      onSuccess,
+      onError,
+    }: {
+      collection: { canisterId: string; standard: NonFungibleStandard };
+      onSuccess: (collectionInfo: CollectionInfo) => void;
+      onError: () => void;
+    },
+    { rejectWithValue }
+  ) => {
+    const instance = KeyRing.getInstance();
+    try {
+      const collectionInfo = await instance?.getNFTInfo(collection);
+      onSuccess(collectionInfo);
+    } catch (e: any) {
+      console.log('Error while fetching Collection info', e);
+      onError?.();
```

Comment:
i noticed that we are using different names for success and error. I started renaming some of them to `onFailure` to unify. what u think of renaming `onError` to `onFailure`? and start using only `onSuccess` and `onFailure`

Reply:
Yes! MB